examples/datadog-express-apm/host-agent/utils/response.js

```diff
@@ -0,0 +1,78 @@
+const agent = require('../agent').default
+/**
+ * @class Response
+ *
+ * @description class to handle all http response
+ */
+class ResponseHandler {
+  /**
+   *
+   * @description method to handle all success responses
+   *
+   * @param response Object
+   * @param statusCode String
+   * @param message String
+   * @param data Object | Array
+   *
+   * @return response JSON
+   */
+  static successResponse = (response, statusCode, message, data) => {
+    const responseBody = { status: 'success' }
+
+    if (message !== '') {
+      responseBody.message = message
+    }
+
+    if (data) {
+      responseBody.data = data
+    }
+
+    agent
+      .init()
+      .dogstatsd.increment(
+        `Commercetools_Client_Response_Success_${statusCode}`,
```

Comment:
I think it is better to have one metric with the name `Commercetools_Client_Response_Success` than multiple metrics with names like `Commercetools_Client_Response_Success_200`, `Commercetools_Client_Response_Success_201`, `Commercetools_Client_Response_Success_202` etc. The response code could be put as a tag. We also did it like this in java: https://github.com/commercetools/commercetools-sdk-java-v2/blob/main/commercetools/commercetools-monitoring-datadog/src/main/java/com/commercetools/monitoring/datadog/DatadogUtils.java#L43

The similar idea is applied also for error counters.

Reply:
Not really, that is not how it works, it will aggregate all the metrics with the same name and increment the count or average it, there are no sign of multiple occurrence of the name `Commercetools_Client_Response_Success`

You can look at the screenshot for reference.

Reply:
But the screenshots are from NewRelic, not datadog. Can you post screenshots from Datadog as well? The difference between datadog and newrelic custom metrics is that in datadog you can add tags (https://docs.datadoghq.com/metrics/custom_metrics/). On the other hand, NewRelic has only time slice metrics, therefore naming convention there is different: https://docs.newrelic.com/docs/apm/agents/manage-apm-agents/agent-data/collect-custom-metrics/

Reply:
I don't know if you can see it, but I can, and there's no duplicate there. If you are in doubt please clone the example repo, run it and show me the duplicate.

<img width="1087" alt="Screenshot 2024-05-27 at 11 09 02" src="https://github.com/commercetools/commercetools-sdk-typescript/assets/32770340/2f589d59-e044-4764-bd53-1c4abae929c1">

Reply:
The image is blurry. But Lam is correct here. The status code should be part of the metrics attributes. This way you can have a total view of all requests and will be able to filter/group the metrics by status code. If it's part of the name it's harder to combine them in one view.

Should be something like this:

```
.dogstatsd.increment(
        `Commercetools_Client_Response`,
        1,
        { env: 'dev', status_code': statusCode, http_method: req.method, success: true  }
      )